From 2bea5c675a181aca639b9509bb8c71d41985dd9c Mon Sep 17 00:00:00 2001
From: Stanislaw Gruszka <sgruszka@redhat.com>
Date: Wed, 10 Dec 2014 10:29:30 -0500
Subject: [wireless] rt2x00: support Ralink 5362

Message-id: <1418207389-7899-17-git-send-email-sgruszka@redhat.com>
Patchwork-id: 101589
O-Subject: [RHEL7.1 PATCH 16/35] rt2x00: support Ralink 5362.
Bugzilla: 1155538
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: John Greene <jogreene@redhat.com>

From: Canek Pelaez Valdes <canek@ciencias.unam.mx>

BZ#1155538

commit ac0372abf8524a7572a9cdaac6495eb2eba20457 upstream.

Signed-off-by: Canek Pelaez Valdes <canek@ciencias.unam.mx>
Signed-off-by: John W. Linville <linville@tuxdriver.com>
Cc: Stanislaw Gruszka <sgruszka@redhat.com>
Cc: Xose Vazquez Perez <xose.vazquez@gmail.com>
Signed-off-by: Luis Henriques <luis.henriques@canonical.com>

Brew:
https://brewweb.devel.redhat.com/taskinfo?taskID=8330268

Testing:
Compiled only.

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/wireless/rt2x00/rt2800.h b/drivers/net/wireless/rt2x00/rt2800.h
index 7cf6081..ebd5625 100644
--- a/drivers/net/wireless/rt2x00/rt2800.h
+++ b/drivers/net/wireless/rt2x00/rt2800.h
@@ -52,6 +52,7 @@
  * RF5592 2.4G/5G 2T2R
  * RF3070 2.4G 1T1R
  * RF5360 2.4G 1T1R
+ * RF5362 2.4G 1T1R
  * RF5370 2.4G 1T1R
  * RF5390 2.4G 1T1R
  */
@@ -72,6 +73,7 @@
 #define RF3070    0x3070
 #define RF3290    0x3290
 #define RF5360    0x5360
+#define RF5362    0x5362
 #define RF5370    0x5370
 #define RF5372    0x5372
 #define RF5390    0x5390
@@ -2145,7 +2147,7 @@ struct mac_iveiv_entry {
 /* Bits [7-4] for RF3320 (RT3370/RT3390), on other chipsets reserved */
 #define RFCSR3_PA1_BIAS_CCK  FIELD8(0x70)
 #define RFCSR3_PA2_CASCODE_BIAS_CCKK FIELD8(0x80)
-/* Bits for RF3290/RF5360/RF5370/RF5372/RF5390/RF5392 */
+/* Bits for RF3290/RF5360/RF5362/RF5370/RF5372/RF5390/RF5392 */
 #define RFCSR3_VCOCAL_EN  FIELD8(0x80)
 /* Bits for RF3050 */
 #define RFCSR3_BIT1   FIELD8(0x02)
diff --git a/drivers/net/wireless/rt2x00/rt2800lib.c b/drivers/net/wireless/rt2x00/rt2800lib.c
index c17fcf2..56e9b58 100644
--- a/drivers/net/wireless/rt2x00/rt2800lib.c
+++ b/drivers/net/wireless/rt2x00/rt2800lib.c
@@ -3141,6 +3141,7 @@ static void rt2800_config_channel(struct rt2x00_dev *rt2x00dev,
   break;
  case RF3070:
  case RF5360:
+ case RF5362:
  case RF5370:
  case RF5372:
  case RF5390:
@@ -3158,6 +3159,7 @@ static void rt2800_config_channel(struct rt2x00_dev *rt2x00dev,
      rt2x00_rf(rt2x00dev, RF3290) ||
      rt2x00_rf(rt2x00dev, RF3322) ||
      rt2x00_rf(rt2x00dev, RF5360) ||
+     rt2x00_rf(rt2x00dev, RF5362) ||
      rt2x00_rf(rt2x00dev, RF5370) ||
      rt2x00_rf(rt2x00dev, RF5372) ||
      rt2x00_rf(rt2x00dev, RF5390) ||
@@ -4272,6 +4274,7 @@ void rt2800_vco_calibration(struct rt2x00_dev *rt2x00dev)
  case RF3070:
  case RF3290:
  case RF5360:
+ case RF5362:
  case RF5370:
  case RF5372:
  case RF5390:
@@ -7072,6 +7075,7 @@ static int rt2800_init_eeprom(struct rt2x00_dev *rt2x00dev)
  case RF3320:
  case RF3322:
  case RF5360:
+ case RF5362:
  case RF5370:
  case RF5372:
  case RF5390:
@@ -7528,6 +7532,7 @@ static int rt2800_probe_hw_mode(struct rt2x00_dev *rt2x00dev)
  case RF3320:
  case RF3322:
  case RF5360:
+ case RF5362:
  case RF5370:
  case RF5372:
  case RF5390:
@@ -7657,6 +7662,7 @@ static int rt2800_probe_hw_mode(struct rt2x00_dev *rt2x00dev)
  case RF3070:
  case RF3290:
  case RF5360:
+ case RF5362:
  case RF5370:
  case RF5372:
  case RF5390:
-- 
1.7.1